src/distillery-api/distillery-api.controller.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  Controller,
+  Get,
+  Param,
+  Body,
+  Put,
+  Post,
+  Delete,
+} from '@nestjs/common';
+import {
+  InjectInMemoryDBService,
+  InMemoryDBService,
+} from '@nestjs-addons/in-memory-db';
+
+import { DistilleryEntity } from '../core';
+
+@Controller('distillery')
+export class DistilleryApiController {
+  constructor(
+    @InjectInMemoryDBService('distillery')
+    private distilleryService: InMemoryDBService<DistilleryEntity>,
+  ) {}
+
+  @Post()
+  async createDistillery(@Body() distillery: Partial<DistilleryEntity>) {
+    return this.distilleryService.create(distillery);
+  }
+
+  @Delete(':id')
+  deleteDistilleryById(@Param('id') id) {
+    return this.distilleryService.delete(id);
+  }
+
+  @Get()
+  getAll(): DistilleryEntity[] {
+    return this.distilleryService.getAll();
+  }
+
+  @Get(':id')
+  getDistilleryById(@Param('id') id) {
+    return this.distilleryService.get(id);
+  }
+
+  @Put(':id')
+  updateDistilleryById(@Param('id') id, @Body() distillery: DistilleryEntity) {
+    id = id * 1; // Convert to number, not the best way to handle this
```

Comment:
Do you have better thoughts here? My brain was not working great last night and this was the quick fix, but I feel like there is a more "proper" way. Dto conversion?

Reply:
Yes, we should create a DTO with `class-transformer` and `class-validator`. I think both of those should be covered in the workshop to ensure students are aware of these techniques. I'm going to merge this PR as is and then we can create an issue/PR for the DTO work.